### DISCUSSION TITLE
Where to store OpenAI key so that the avante plugin can auto load it when opened.

### DISCUSSION THREAD

**Discussion by sanjeethboddi** [yetone/avante.nvim#1004] [Q&A]
Hi. 

I do not want to store my api secrets in .zshrc/.bashrc. Are there other options which i can follow?

**Comment by tom-kuca** [comment]
Yes, see [wiki](https://github.com/yetone/avante.nvim/wiki).

Here a snippet that gets the password from password manager `pass`.
```lua
		claude = {
			api_key_name = { "pass", "api/claude" },
			temperature = 0,
			max_tokens = 2048,
		},
```

**Comment by einarpersson** [comment]
```
		claude = {
			api_key_name = { "cat", "/path/to/file.key" },
		},
```

if you just want to get it from a file somewhere... (and of course it is better to have all these things in some password manager but...)
